dds_web/database/models.py

```diff
@@ -130,7 +122,42 @@ class User(db.Model):
     def __repr__(self):
         """Called by print, creates representation of object"""
 
-        return f"<User {self.username}>"
+        return f"<User {self.username}:{self.role}>"
+
+
+class ResearchUser(User):
+    username = db.Column(db.String(20), db.ForeignKey("users.username"), primary_key=True)
+    # One user can have many projects, and one projects can have many users
+    projects = db.relationship(
+        "Project", secondary=project_users, backref=db.backref("users", lazy="dynamic")
+    )
+
+    __mapper_args__ = {
+        "polymorphic_identity": "research_user",
+    }
+
+    @property
+    def role(self):
+        return "researcher"
+
+
+class UnitUser(User):
+    username = db.Column(db.String(20), db.ForeignKey("users.username"), primary_key=True)
+    admin = db.Column(db.Boolean, unique=False, nullable=False)
+    # Foreign keys
+    # One facility can have many users
+    facility_id = db.Column(db.Integer, db.ForeignKey("facilities.id"))
+
+    __mapper_args__ = {
+        "polymorphic_identity": "unit_user",
+    }
+
+    @property
+    def role(self):
+        if self.admin:
+            return "unit_admin"
+        else:
+            return "unit_user"
 
```

Comment:
Would there be a SuperAdminUser class inheriting from User? If not how do you think of it?

Reply:
Yeah I think I would do it that way.

Reply:
I agree.

Reply:
+1